cadence/contracts/connectors/evm/UniswapV3SwapConnectors.cdc

```diff
@@ -0,0 +1,779 @@
+                    let res: EVM.Result? = self._callRaw(
+                        to: to,
+                        calldata: data,
+                        gasLimit: 1_500_000,
+                        value: 0
+                    )
+                    return res!.data
+                }
+
+                let comp: Int = compressed(currentTick, tickSpacing)
+                let baseWord: Int = comp >> 8 // 256 ticks per word
+
+                var result: [{String: Int}] = []
+                var w = -wordRadius
+                while w <= wordRadius {
+                    let wordIndex = baseWord + w
+
+                    // tickBitmap(int16)
+                    let tbData = evmCallBytes(pool, encode1Int(SEL_TICK_BITMAP, wordIndex, 16))
+                    let tbWord = wordToUInt(words(tbData)[0])
+                    if tbWord != 0 {
+                        var bit: Int = 0
+                        while bit < 256 {
+                            let mask: UInt = UInt(1) << UInt(bit)
+                            if (tbWord & mask) != 0 {
+                                let tickIndex: Int = (wordIndex << 8) + bit
+                                let popped: Int = tickIndex * tickSpacing
+
+                                // ticks(int24)
+                                let txBytes = evmCallBytes(pool, encode1Int(SEL_TICKS, popped, 24))
+                                let ws = words(txBytes)
+                                let liqGross = wordToUIntN(ws[0], 128)
+                                let liqNet   = wordToIntN(ws[1], 128)
+                                if liqGross != 0 {
+                                    result.append({ "tick": popped, "liqNet": liqNet })
+                                }
+                            }
+                            bit = bit + 1
+                        }
+                    }
+                    w = w + 1
+                }
+
+                // insertion sort by tick ASC
+                var i = 1
+                while i < result.length {
+                    let key = result[i]
+                    var j = i - 1
+                    while j >= 0 && result[j]["tick"]! > key["tick"]! {
+                        result[j + 1] = result[j]
+                        j = j - 1
+                    }
+                    result[j + 1] = key
+                    i = i + 1
+                }
+
+                return result
+            }
+
+            //
+            // --- Selectors (locals, not top-level) ---
+            //
+            let SEL_SLOT0: [UInt8]        = [0x38, 0x50, 0xc7, 0xbd]
+            let SEL_LIQUIDITY: [UInt8]    = [0x1a, 0x68, 0x65, 0x02]
+            let SEL_FEE: [UInt8]          = [0xdd, 0xca, 0x3f, 0x43]
+            let SEL_TICK_BITMAP: [UInt8]  = [0x53, 0x39, 0xc2, 0x96]
+            let SEL_TICKS: [UInt8]        = [0xf3, 0x0d, 0xba, 0x93]
+            let SEL_TICK_SPACING: [UInt8] = [0xd0, 0xc9, 0x3a, 0x7c]
```

Comment:
Same comment as elsewhere - it's difficult to catch any mismatch against the desired selector. Suggest decoding the signature for each using EVM.encodeABIWithSignature() using empty data array.